src/Conversion/InterventionImageFileConverter.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+namespace SilverStripe\Assets\Conversion;
+
+use Intervention\Image\Exception\ImageException;
+use SilverStripe\Assets\Image_Backend;
+use SilverStripe\Assets\InterventionBackend;
+use SilverStripe\Assets\Storage\AssetStore;
+use SilverStripe\Assets\Storage\DBFile;
+use SilverStripe\Core\Injector\Injector;
+
+/**
+ * File converter powered by the Intervention Image library.
+ * Supports any file conversions that Intervention Image can perform.
+ */
+class InterventionImageFileConverter implements FileConverter
+{
+    public function supportsConversion(string $fromFormat, string $toFormat, array $options = []): bool
+    {
+        $unsupportedOptions = $this->validateOptions($options);
+        if (!empty($unsupportedOptions)) {
+            return false;
+        }
+        // This converter requires intervention image as the image backend
+        $backend = Injector::inst()->get(Image_Backend::class);
+        if (!is_a($backend, InterventionBackend::class)) {
+            return false;
+        }
+        return $this->supportedByIntervention($fromFormat, $backend) && $this->supportedByIntervention($toFormat, $backend);
+    }
+
+    public function convert(DBFile $from, string $toFormat, array $options = []): DBFile
+    {
+        // Do some basic validation up front for things we know aren't supported
+        $problems = $this->validateOptions($options);
+        if (!empty($problems)) {
+            throw new FileConversionException('Invalid options provided: ' . implode(', ', $problems));
+        }
+        $originalBackend = $from->getImageBackend();
+        if (!is_a($originalBackend, InterventionBackend::class)) {
+            $actualClass = $originalBackend ? get_class($originalBackend) : 'null';
+            throw new FileConversionException("ImageBackend must be an instance of InterventionBackend. Got $actualClass");
+        }
+        if (!$this->supportedByIntervention($toFormat, $originalBackend)) {
+            throw new FileConversionException("Convertion to format '$toFormat' is not suported.");
+        }
+
+        $quality = $options['quality'] ?? null;
+        // Clone the backend if we're changing quality to avoid affecting other manipulations to that original image
+        $backend = $quality === null ? $originalBackend : clone $originalBackend;
+        // Pass through to invervention image to do the conversion for us.
+        try {
+            $result = $from->manipulateExtension(
```

Comment:
```suggestion
            $to = $from->manipulateExtension(
```

Reply:
This is very subjective. To me `$result` makes a lot more sense, because it is the result of the conversion. `$to` as a variable name to me implies that the conversion has not yet happened, and this variable stores information about what the conversion will be made *to*.

With that context, if you still insist on me making the change, I'll make it at that point purely to avoid additional ping pong - but I am resisting for now because 
1. it's a subjective change that we have different views on - there is no objectively better name here
2. I prefer the name I already am using for the reasons discussed above